### ISSUE TITLE
ani.json Lese und Schreibe Problem

### ISSUE THREAD

**Issue by Crypto90** [Pfuenzle/anime-loads#35]
Könnte man die Funktion der ani.json für lesen/schreiben verbessern?

Ich bin auf folgendes Problem gestoßen:

Läuft das Processing im Loop an und man fügt in der Zeit manuell etwas neues zum Monitoring (in die ani.json) hinzu, fliegt der hinzugefügte Eintrag wieder raus, wenn im Processing Loop etwas hinzugefügt wird.
Das liegt wohl daran, dass zu Beginn des Processings die ani.json eingelesen wird und bis zum Ende vom Processing diese geöffnet bleibt und aktualisiert wird.
Dabei gehen jedoch Einträge die in der Processing Zeit hinzugefügt werden verloren und man kann das ganze widerholen.

Sinnvoller wäre es, wärend des Processings, wenn etwas aktualisiert werden muss innerhalb der ani.json, zum Zeitpunkt der nötigen Aktualisierung die ani.json neu einzulesen, die Änderungen zu schreiben und dann wieder zu schließen.

So könnten auch neue Einträge hinzugefügt werden, während parallel das Processing läuft. Die Wahrscheinlichkeit für einen Conflict wäre so sehr viel geringer.

**Comment by Pfuenzle** [comment]
Ja, das würde tatsächlich mehr Sinn machen, ist auch geplant bei dem Rewrite, der seit längerem langsam in Arbeit ist.
Der AniBot wurde eigentlich auch eher als Proof of Concept geschrieben und ist dann mit der Zeit gewachsen, also sind viele Dinge darin eher so semi-logisch.

In der aktuellen Version werde ich nichts dran ändern, aber wenn ich mal die neue Version fertig habe, sollte es das problem nicht mehr geben.

**Comment by Crypto90** [comment]
Okay, super gut zu wissen. Danke.

Ich hab dir gerade mal eine Email geschrieben, zum Thema Web Frontend/Webinterface.
